### PR TITLE
Added AuthenticationErrorDetail

### PR DESCRIPTION
**Pull request by tanyasethi-msft**
- [x] The purpose of this PR is explained in this or a referenced issue.
- [x] The PR does not update generated files.
   - These files are managed by the codegen framework at [Azure/autorest.go][].
- [x] Tests are included and/or updated for code changes.
- [x] Updates to module CHANGELOG.md are included.
- [x] MIT license headers are included in each file.

[Azure/autorest.go]: https://github.com/Azure/autorest.go

### CODE REVIEW

File: sdk/storage/azfile/share/client_test.go [hunk 1]
```diff
@@ -278,6 +279,60 @@ func (s *ShareRecordedTestsSuite) TestShareCreateNilMetadata() {
...
+	shareClientSAS, err := share.NewClientWithNoCredential(shareSAS, nil)
+	_require.NoError(err)
+
+	dirClient := testcommon.GetDirectoryClient(testcommon.GenerateDirectoryName(testName), shareClientSAS)
+	_, err = dirClient.Create(context.Background(), nil)
+	_require.Error(err)
+
+	var responseErr *azcore.ResponseError
+	_require.ErrorAs(err, &responseErr)
+	_require.Equal("AuthenticationFailed", responseErr.ErrorCode)
```
Comment (jhendrixMSFT) [inline sdk/storage/azfile/share/client_test.go:332]: `fileerror.AuthenticationFailed` for the error code?

Comment (tanyasethi-msft) [reply sdk/storage/azfile/share/client_test.go]: corrected the above

**Review by souravgupta-msft — APPROVED** [review]

**Pull request merged**